src/prometheus/definition.go

```diff
@@ -701,3 +724,41 @@ func getRawEntityID(parentGroupLabel, groupLabel, entityID string, groups defini
 	}
 	return rawEntityID, nil
 }
+
+func WithLabelValueInMetricNameAndWithLabelValueMappedToMetricValue(rawMetricName, label, format, valueLabel string, valueMap map[string]int) definition.FetchFunc {
+	return func(groupLabel, entityID string, rawGroups definition.RawGroups) (definition.FetchedValue, error) {
+		fetchedValues := definition.FetchedValues{}
+		metricsRaw, err := definition.FromRaw(rawMetricName)(groupLabel, entityID, rawGroups)
+		if err != nil {
+			return nil, fmt.Errorf("fetching metric %q for entity %q in group %q: %w", rawMetricName, entityID, groupLabel, err)
+		}
+
+		metrics, ok := metricsRaw.([]Metric)
+		if !ok {
+			return nil, fmt.Errorf("incompatible metric type for %q. Expected: []Metric. Got: %T", rawMetricName, metricsRaw)
+		}
+
+		for _, metric := range metrics {
+			nameLabelValue, ok := metric.Labels[label]
+			if !ok {
+				return nil, fmt.Errorf("label %q for metric name not found in metric %v", label, metric)
+			}
+
+			fetchedValueName := fmt.Sprintf(format, nameLabelValue)
+
+			valueLabelValue, ok := metric.Labels[valueLabel]
+			if !ok {
+				return nil, fmt.Errorf("label %q for metric value not found in metric %v", valueLabel, metric)
+			}
+
+			metricValue, ok := valueMap[valueLabelValue]
+			if !ok {
+				return nil, fmt.Errorf("key %q not found in value map %v", valueLabelValue, valueMap)
+			}
+
+			fetchedValues[fetchedValueName] = metricValue
+		}
+
+		return fetchedValues, nil
```

Comment:
If we switch to a more lenient parsing as I suggested, we would need to add a non-zero length here in case we were not able to parse any metric and return an error.